### PR TITLE
Grouped 3d conv backward data support

### PR DESCRIPTION
**Pull request by bartekxk**

### CODE REVIEW

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_data/device_grouped_conv_bwd_data_xdl_instance.hpp [hunk 1]
```diff
@@ -40,100 +34,106 @@ static constexpr auto ConvBwdDataFilter1x1Stride1Pad0 =
...
+        // DeviceGroupedConvBwdDataMultipleD_Xdl_CShuffle_v1<       3,   GNHWK,   GKYXC, Empty_Tuple,   GNHWC,   F16,   F16,     F32,      F16, Empty_Tuple,   F16,  PassThrough,  PassThrough,    PassThrough,  ConvBwdDataDefault,  true,  true,             1,   256,   256,   128,    32,   8,   8,   32,   32,       4,       2,       S<4, 64, 1>,     S<1, 0, 2>,     S<1, 0, 2>,              2,              8,              8,         1,       S<4, 64, 1>,     S<1, 0, 2>,     S<1, 0, 2>,              1,              8,              8,         1,            1,            1,     S<1, 32, 1, 8>,                8>,
```
Comment (aosewski) [inline library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_data/device_grouped_conv_bwd_data_xdl_instance.hpp]: Why are you changing here 2D -> 3D ?

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_backward_data.hpp [hunk 2]
```diff
@@ -100,6 +100,91 @@ void add_device_grouped_conv2d_bwd_data_xdl_nhwgc_gkyxc_nhwgk_bf16_instances(
                                                                   PassThrough,
                                                                   PassThrough>>>& instances);
 
+// conv3d backward data
+void add_device_grouped_conv3d_bwd_data_xdl_gndhwc_gkzyxc_gndhwk_f16_instances(
```
Comment (aosewski) [inline library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_backward_data.hpp]: The name of this function is not consistent with the Kernel input data layout order. There is: in, wei, out and should be out, wei, in

**Review by aosewski — COMMENTED** [review]
In general a lot of good work! :) Have just two minor comments/questions.

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_data/device_grouped_conv_bwd_data_xdl_instance.hpp (hunk 1 above)
Comment (bartekxk) [reply library/include/ck/library/tensor_operation_instance/gpu/grouped_conv_bwd_data/device_grouped_conv_bwd_data_xdl_instance.hpp]: Done.

File: library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_backward_data.hpp (hunk 2 above)
Comment (bartekxk) [reply library/include/ck/library/tensor_operation_instance/gpu/grouped_convolution_backward_data.hpp]: Done.

**Review by aosewski — APPROVED** [review]

**Pull request merged**